zppy/utils.py

```diff
@@ -140,9 +143,135 @@ def getComponent(input_files):
 
 
 # -----------------------------------------------------------------------------
-def submitScript(scriptFile, dependFiles=[], export="ALL"):
+# Bundles
+class Bundle(object):
+    def __init__(self, c):
+        self.bundle_name: str = c["bundle"]
+        self.dry_run: bool = c[
+            "dry_run"
+        ]  # Value is taken from the first task in the bundle
+        self.script_dir: str = c[
+            "scriptDir"
+        ]  # Value is taken from the first task in the bundle
+
+        self.bundle_file: str = f"{self.script_dir}/{self.bundle_name}.bash"
+        self.bundle_output: str = self.bundle_file.strip(".bash") + ".o%j"
+
+        self.dependencies_in_bundle_file: Set[str] = set()
+        self.dependencies_not_in_bundle_file: Set[str] = set()
+
+        self.export: str = "NONE"
+
+    def create_header(self, scriptFile):
+        if os.path.exists(self.bundle_file):
+            error_message = f"create_header is being applied to an existing bundle: {self.bundle_name}. If {self.bundle_name} was running and you re-ran zppy, that is the likely cause of this error. If {self.bundle_name} is not currently running, delete {self.bundle_file} and run zppy again."
+            # By failing, zppy is prevented from restarting / overwriting work the bundle is currently doing.
+            raise FileExistsError(error_message)
+
+        # Create header or structure to keep info for the header.
+        with open(self.bundle_file, "w") as destination_file:
+            destination_file.write("#!/bin/bash\n")
+            with open(scriptFile, "r") as source_file:
+                for line in source_file:
+                    # Copy to bundle_file all lines from script_file that contain "SBATCH"
+                    if re.search("SBATCH", line):
+                        if re.search("--output", line):
+                            destination_file.write(
+                                f"#SBATCH --output={self.bundle_output}\n"
+                            )
+                        elif re.search("--job-name", line):
+                            destination_file.write(
+                                f"#SBATCH --job-name={self.bundle_name}\n"
```

Comment:
We want to be sure to name the job with the bundle's name, not the first task's name.